### ISSUE TITLE
Groupby doesn't call aggregation on empty groups

### ISSUE THREAD

**Issue by TomAugspurger** [pandas-dev/pandas#18869]
This should exit the interpreter.

```python
In [1]: import pandas as pd

In [2]: df = pd.DataFrame({"A": pd.Categorical(['a', 'a'], categories=['a', 'b']),
   ...:                    "B": [1, 1]})
   ...:

In [3]: import sys

In [4]: def f(x):
   ...:     if len(x) == 0:
   ...:         sys.exit(1)
   ...:     return len(x)
   ...:

In [5]: df.groupby('A').agg(f)
Out[5]:
     B
A
a  2.0
b  NaN
```

Instead, I think groupby assumes the output of the custom aggfunc is `NaN`.

**Comment by arw2019** [comment]
This is still a problem on 1.2 master

**Comment by rhshadrach** [comment]
This is fixed on main, could use tests.

**Comment by HaruguchiKazuto** [comment]
@rhshadrach 
Hello, I would like to work on this issue. This will be my first contribution to oss.
I understand that I can add test cases.
If it's convenient, could you please let me know which file I should add the test cases to? That would be very helpful.

**Comment by rhshadrach** [comment]
Thanks @HaruguchiKazuto - this would go in `pandas.tests.groupby.aggregate.test_aggregate`.

**Comment by HaruguchiKazuto** [comment]
take

**Comment by rhshadrach** [comment]
Closed by #56145
